libraries/rib-base/src/main/java/com/badoo/ribs/clienthelper/childawareness/ChildAwareRegistry.kt

```diff
@@ -0,0 +1,29 @@
+package com.badoo.ribs.clienthelper.childawareness
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.plugin.NodeAware
+import com.badoo.ribs.core.plugin.SubtreeChangeAware
+import kotlin.reflect.KClass
+
+/**
+ * This interface provides simplified access to the Nodes children
+ * when you need to create connection between them without any intermediate processing.
+ */
+interface ChildAwareRegistry : SubtreeChangeAware, NodeAware {
+
+    fun <T : Rib, R : Rib> whenChildrenAttached(
```

Comment:
I believe R usually refers to a Result type, so here it would be T1 & T2

```suggestion
    fun <T1 : Rib, T2 : Rib> whenChildrenAttached(
```